blockmanager.go

```diff
@@ -873,6 +913,36 @@ func (b *blockManager) handleInvMsg(imsg *invMsg) {
 		}
 	}
 
+	// If this inv contains a block annoucement, and this isn't coming from
+	// our current sync peer or we're current, then update the last
+	// announced block for this peer. We'll use this information later to
+	// update the heights of peers based on blocks we've accepted that they
+	// previously announced.
+	if lastBlock != -1 && (imsg.peer != b.syncPeer || b.current()) {
+		imsg.peer.UpdateLastAnnouncedBlock(&invVects[lastBlock].Hash)
+	}
+
+	// Ignore invs from peers that aren't the sync if we are not current.
+	// Helps prevent fetching a mass of orphans.
+	if imsg.peer != b.syncPeer && !b.current() {
+		return
+	}
+
+	// If our chain is current and a peer announces a block we already
+	// know of, then update their current block height.
+	if lastBlock != -1 && b.current() {
+		exsits, err := b.server.db.ExistsSha(&invVects[lastBlock].Hash)
+		if err == nil && exsits {
+			blkHeight, err := b.server.db.FetchBlockHeightBySha(&invVects[lastBlock].Hash)
+			if err != nil || blkHeight == 0 {
```

Comment:
Agreed that the check for 0 is unnecessary, and can result in an unhelpful error log similar to a prior case in involving the `addrindex`. I was previously also checking for zero since in the case of an error, `db. FetchBlockHeightBySha` will return `0` for the height *and* and a non-nil error: 

```go
func (db *LevelDb) getBlkLoc(sha *wire.ShaHash) (int64, error) {
	key := shaBlkToKey(sha)

	data, err := db.lDb.Get(key, db.ro)
	if err != nil {
		if err == leveldb.ErrNotFound {
			err = database.ErrBlockShaMissing
		}
		return 0, err
	}
...
```

However, since the cases are coupled together it's sufficient to only check for a non-nil error. I've updated this fragment to only check for the error.